src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -432,7 +433,10 @@ public bool TryCreateType(ITypeSymbol symbol, Func<System.Type, bool> validator,
 
             if (existingType != null && validator(existingType))
             {
-                var arguments = namedTypeSymbol.TypeArguments.Select(a => CreateType(a)).ToArray();
+                if (!TryPopulateArguments(namedTypeSymbol.TypeArguments, out var arguments))
```

Comment:
I believe the `validator` delegate should also pass through here.

Reply:
Not following here.

Reply:
See what you mean now, passed along the validator